File: Day354/RangeSumOfBST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sum;
    public void preorder(TreeNode root, int low, int high) {
        if (root == null) {
            return;
        }
        if (root.val <= high && root.val >= low) {
            sum += root.val;
        }
        preorder(root.left, low, high);
        preorder(root.right, low, high);
    }
    public int rangeSumBST(TreeNode root, int low, int high) {
        sum = 0;
        preorder(root, low, high);
        return sum;
    }
}

File: Day354/RangeSumOfBST.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

 func inorder(root *TreeNode, sum *int, low int, high int) {
    if root == nil {
        return
    }
    inorder(root.Left,sum, low, high)
    // *list = append(*list, root.Val)
    if root.Val >= low && root.Val <= high {
        *sum += root.Val
    }
    inorder(root.Right, sum, low, high)
}
func rangeSumBST(root *TreeNode, low int, high int) int {
    sum := 0
    // list := make([]int, 0)
    inorder(root,  &sum, low, high)
    // fmt.Println(list)
    return sum
}